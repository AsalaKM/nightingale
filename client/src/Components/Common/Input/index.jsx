import React from "react";
import propTypes from "prop-types";
import { Lable, Input } from "./index.style";

export default function index(props) {
  const { type, label, value, onChange, name, width, placeholder } = props;
  return (
    <>
      <Lable htmlFor={name}>
        {label}{" "}
        <Input
          name={name}
          type={type}
          style={{ width }}
          placeholder={placeholder}
          value={value}
          onChange={onChange}
          {...props}
        />
      </Lable>
    </>
  );
}
index.propTypes = {
  value: propTypes.string,
  onChange: propTypes.func,
  name: propTypes.string.isRequired,
  label: propTypes.string,
  type: propTypes.string,
  width: propTypes.string
};
