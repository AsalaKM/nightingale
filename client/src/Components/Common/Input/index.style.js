
import styled from 'styled-components'

export const Lable = styled.label`
    display: block;
    font-size:15px;
    font-weight: 500;
    color:#333;
    margin: 1rem;
`

export const Input = styled.input`
    outline:none;
    border: solid .5px #C8C7CC;
    margin: .5rem 0;
    display: block;
    height: 35px;
    padding: 0 1rem;
    border-radius: 100px;
    outline-color: orange;
    font-size: 16px;
`