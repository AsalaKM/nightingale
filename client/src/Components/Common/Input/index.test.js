import React from "react";
import renderer from "react-test-renderer";
import Input from "./index.jsx";

describe("Input", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Input label="Username" name="username" type="text" width="20rem" />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
