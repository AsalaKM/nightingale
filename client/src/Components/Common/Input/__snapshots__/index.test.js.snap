// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<label
  className="sc-bdVaJa eWdFUc"
  htmlFor="username"
>
  Username
   
  <input
    className="sc-bwzfXH eZzzSf"
    label="Username"
    name="username"
    style={
      Object {
        "width": "20rem",
      }
    }
    type="text"
    width="20rem"
  />
</label>
`;
